src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -361,5 +392,16 @@ private static bool IsApplicationRelativePath(string name)
             Debug.Assert(!string.IsNullOrEmpty(name));
             return name[0] == '~' || name[0] == '/';
         }
+
+        private static bool IsRelativePath(string name)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+
+            // **Reviewers**: Though ./ViewName looks like a relative path, checking for '/' leads to problems with
```

Comment:
I think of the view component case as _using_ a feature that's available to everyone rater than as a unique requirement. Further view searches with slashes in the name work in MVC 5.x.y.